sale_double_validation/models/sale.py

```diff
@@ -1,22 +1,37 @@
 # -*- coding: utf-8 -*-
 # Copyright 2017 Camptocamp SA
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
-from odoo import api, fields, models
+from odoo import api, models
+from odoo.tools import float_compare
 
 
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    state = fields.Selection(
-        selection_add=[('to_approve', 'To Approve')]
-    )
+    @api.model
+    def _setup_fields(self, partial):
+        super(SaleOrder, self)._setup_fields(partial)
+        selection = self._fields['state'].selection
+        position = 0
+        exists = False
+        for idx, (state, __) in enumerate(selection):
+            if state == 'draft':
+                position = idx
+            elif state == 'to_approve':
+                exists = True
+        if not exists:
+            selection.insert(position + 1, ('to_approve', _('To Approve')))
 
     @api.multi
     def is_amount_to_approve(self):
         currency = self.env.user.company_id.currency_id
         limit_amount = self.company_id.so_double_validation_amount
         limit_amount = currency.compute(limit_amount, self.currency_id)
-        return self.amount_total >= limit_amount
+        if float_compare(
+           self.amount_total, limit_amount, precision_digits=2) == 1:
```

Comment:
You can't hardcode the precision_digits which depends of the currency. 
Should be `precision_rounding=self.currency_id.rounding`.

Another point: you could shorten by returning the boolean result `return (float_compare... == 1)`